### PR TITLE
Rename and cleanup some members in the 'use var' feature.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Precursor to some additional work i'm doing here.

Reviewing a commit at a time would likely be useful.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @jcouv

File: src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractMethodTests.cs [hunk 1]
```diff
@@ -1423,7 +1423,7 @@ private static string NewMethod(int i)
 
         return v;
     }
-}", options: Option(CSharpCodeStyleOptions.UseImplicitTypeForIntrinsicTypes, CodeStyleOptions.TrueWithSuggestionEnforcement));
+}", options: Option(CSharpCodeStyleOptions.VarForBuiltInTypes, CodeStyleOptions.TrueWithSuggestionEnforcement));
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractMethodTests.cs]: these are the names we use for our actual editorconfig settings.  i'm trying to unify them all so we can have a consistent way of talking about the options from the UI, to editorconfig, to our internal enums/options.

File: src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs [hunk 2]
```diff
@@ -5,45 +5,13 @@
...
-
-        /// <summary>
-        /// Given an expression of assignment, answers if its type is
-        /// considered an intrinsic predefined type.
-        /// </summary>
-        private static bool IsBuiltInType(
-            ExpressionSyntax initializerExpression,
-            SemanticModel semanticModel,
-            CancellationToken cancellationToken) =>
-                IsBuiltInType(semanticModel.GetTypeInfo(initializerExpression, cancellationToken).Type);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs]: these methods were never called.

File: src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs [hunk 3]
```diff
@@ -5,45 +5,13 @@
...
-            SemanticModel semanticModel,
-            OptionSet optionSet,
-            CancellationToken cancellationToken)
-        {
-            var stylePreferences = GetCurrentTypeStylePreferences(optionSet);
-
-            var isBuiltInTypeContext = IsBuiltInType(initializerExpression, semanticModel, cancellationToken);
-
-            var isTypeApparentContext = IsTypeApparentInAssignmentExpression(
-                stylePreferences, initializerExpression, semanticModel, cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs]: this was the only caller of IsTypeApparentInAssignmentExpression that used the defualt argument.

File: src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs [hunk 4]
```diff
@@ -75,19 +43,34 @@ public static bool IsBuiltInType(ITypeSymbol type)
...
+                }
+
+                for (int i = 0, n = tuple.Arguments.Count; i < n; i++)
                 {
-                    if (!IsTypeApparentInAssignmentExpression(stylePreferences, argument.Expression, semanticModel, cancellationToken, typeInDeclaration))
+                    var argument = tuple.Arguments[i];
+                    var tupleElementType = tupleType.TupleElements[i].Type;
+
+                    if (!IsTypeApparentInAssignmentExpression(
+                            stylePreferences, argument.Expression, semanticModel, tupleElementType, cancellationToken))
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs]: logic here appeared to be busted.  we continued to pass in the topmost tuple type when asking if each tuple expr was apparant or not.

**Comment by CyrusNajmabadi** [comment]
@ryzngard can i get a review plz?  Thanks much!

File: src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyle.cs [hunk 5]
```diff
@@ -11,11 +11,12 @@
...
+    internal enum UseVarPreference
     {
         None = 0,
-        ImplicitTypeForIntrinsicTypes = 1 << 0,
-        ImplicitTypeWhereApparent = 1 << 1,
-        ImplicitTypeWherePossible = 1 << 2,
+        ForBuiltInTypes = 1 << 0,
+        WhenTypeIsApparent = 1 << 1,
+        Elsewhere = 1 << 2,
+        // ImplicitTypeWhereExplicit = 1 << 3,
```
Comment (ryzngard) [inline src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyle.cs]: remove?

File: src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs (hunk 4 above)
Comment (ryzngard) [reply src/Workspaces/CSharp/Portable/CodeStyle/TypeStyle/TypeStyleHelper.cs]: Is there a way to add a test for this? I agree this looks more correct but having a test to verify the behavior would be nice :)

**Comment by CyrusNajmabadi** [comment]
@ryzngard Anything else you want here?  Can you merge in?

**Comment by CyrusNajmabadi** [comment]
Thanks

**Pull request merged**